### PR TITLE
Port scenario_intersection to ROS2

### PR DESCRIPTION
**Pull request by nnmm**
Copied from @jilaada's PR https://github.com/tier4/scenario_runner.iv.universe/pull/13 where most of the work has happened:
> Relatively straightforward port of the scenario_intersection package. Some changes are not complete yet due to unmet dependencies.
> 
> Notable Changes:
>     * ros::Time is passed in as a method but not used - this was removed
>     * rosbag package was listed as a dependency in the package.xml - this was not seen anywhere and therefore removed
>     * Rearranged headers so they were only where they were needed

I only uncommented the relevant code and fixed all compiler errors on a branch where I merged in the `scenario_api` PR. I also changed the logger back to be defined by the package itself instead of requiring an external logger. I think passing it in would complicate `scenario_runner` a bit, which constructs these classes.

### CODE REVIEW

File: entity/scenario_intersection/include/scenario_intersection/intersection.h [hunk 1]
```diff
@@ -94,6 +96,7 @@ class Intersection
                     target_, boost::lexical_cast<std::string>(each), false);
               });
         }
+        return false;
```
Comment (nnmm) [inline entity/scenario_intersection/include/scenario_intersection/intersection.h:99]: Here a return value was missing. I added `false` but am not sure if it's correct. @yamacir-kit as the author of this package, could you let me know what this value should be?

Comment (yamacir-kit) [reply entity/scenario_intersection/include/scenario_intersection/intersection.h:99]: @nnmm Your change is correct :+1:

**Review by jilaada — COMMENTED** [review]
Thanks for finishing this package and fixing the logger implementation!

**Comment by mitsudome-r** [comment]
@cvasfi Please review this PR.

File: entity/scenario_intersection/include/scenario_intersection/intersection.h (hunk 1 above)
Comment (nnmm) [reply entity/scenario_intersection/include/scenario_intersection/intersection.h:99]: Thank you!

File: entity/scenario_intersection/include/scenario_intersection/utility.h [hunk 2]
```diff
@@ -14,7 +15,7 @@ template <typename T>
 T convert(const std::string&);
 
 template <typename T, typename F>
-decltype(auto) if_exist(const YAML::Node& node, const std::string& key, F&& consequent)
+decltype(auto) if_exist(const YAML::Node& node, const std::string& key, F&& consequent, rclcpp::Logger & logger)
```
Comment (cvasfi) [inline entity/scenario_intersection/include/scenario_intersection/utility.h]: Why not have this: 
```suggestion
decltype(auto) if_exist(const YAML::Node& node, const std::string& key, F&& consequent, const rclcpp::Logger & logger)
```

Same with `Controller`,

**Review by cvasfi — CHANGES_REQUESTED** [review]

File: entity/scenario_intersection/include/scenario_intersection/utility.h (hunk 2 above)
Comment (nnmm) [reply entity/scenario_intersection/include/scenario_intersection/utility.h]: Done

**Comment by nnmm** [comment]
@cvasfi Do you have further comments? Otherwise @esteve could merge this and we can move on to the next group of packages.

**Review by cvasfi — APPROVED** [review]

**Comment by cvasfi** [comment]
@nnmm LGTM.

**Review by esteve — APPROVED** [review]

**Pull request merged**